regression/test.pl

```diff
@@ -104,6 +105,19 @@ ($$$$$$$$$)
     }
   }
 
+  # Exclude tests that have an OS- tag that doesn't match the current
+  # architecture
+  my $os;
+  if($^O eq "linux") { $os = "LINUX" }
+  if($^O eq "cygwin") { $os = "WINDOWS" }
+  if($^O eq "msys") { $os = "WINDOWS" }
+  if($^O eq "MSWin32") { $os = "WINDOWS" }
+  if($^O eq "darwin") { $os = "MACOS" }
+
+  if(not exists($tags_set{"OS-" . $os}) and grep { "OS-" eq substr($_, 0, length("OS-")) } @tags) {
+    $passes_tag_tests = 0;
+  }
+
```

Comment:
I'm not confident about this one. The surrounding environment has already solved this problem: CMake surely comes up with some identifier of the operating system (here, the crucial one will be `WIN32`), and we've got a solution for Make in `src/common`.